vms/platformvm/state/state.go

```diff
@@ -1609,7 +1609,7 @@ func (s *state) writeCurrentStakers(updateValidators bool, height uint64) error
 					PotentialDelegateeReward: 0,
 				}
 
-				metadataBytes, err := blocks.GenesisCodec.Marshal(blocks.Version, metadata)
+				metadataBytes, err := validatorMetadataCodec.Marshal(validatorMetadataCodecV0, metadata)
```

Comment:
should have I not changed the `v0tag` content from "serialize" to "v0", I would not have spotted this and left `blocks.Version` here (as I did in previous commit. I think we should avoid using the detatching codecs